content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+
+Clients which are only using atproto OAuth for account authentication (without authorization to access PDS resources) should request minimal scopes (see “Scopes” section), but still need to implement most of the authorization flow. In particular, they must use the `sub` field in a token response to verify the account identity (this is an atproto-specific detail).
+
+#### Client Metadata
+
+:::note
+The client metadata standard is still a draft and may evolve over time. Our intention is to evolve and align with subsequent drafts and any final standard, while minimizing disruption and breakage with existing implementations.
+:::
+
+Clients must publish a "client metadata" JSON file on the public web. This will be fetched dynamically by Authorization Servers as part of the authorization request (PAR) and at other times during the session lifecycle. The response HTTP status must be 200 (not another 2xx or a redirect), with a JSON object body with the correct `Content-Type` (`application/json`).
+
+Authorization Servers need to fetch client metadata documents from the public web. They should use a hardened HTTP client for these requests (see "OAuth Security Considerations"). Servers may cache client metadata responses, optionally respecting HTTP caching headers (within limits). Minimum and maximum cache TTLs are not currently specified, but should be chosen to ensure that auth token requests predicated on stale confidential client authentication keys (`jwks` or `jwks_uris`) are rejected in a timely manner.
+
+The following fields are relevant for all client types:
+
+- `client_id` (string, required): the `client_id`. Must exactly match the full URL used to fetch the client metadata file itself
+- `application_type` (string, optional): must be one of `web` or `native`, with `web` as the default if not specified
+- `grant_types` (array of strings, required): `authorization_code` must always be included. `refresh_token` is optional, but must be included if the client will make token refresh requests.
+- `scope` (string, sub-strings space-separated, required): all scope values which *might* be requested by this client are declared here. The `atproto` scope is required, so must be included here. See “Scopes” section.
+- `response_types` (array of strings, required): `code` must be included.
+- `redirect_uris` (array of strings, required): at least one redirect URI is required. See Authorization Request Fields section.
```

Comment:
- Redirect uris should not contain the port number if it is the default port number for the protocol (`http://x:80`, `https://x:443` are not allowed)
- `http://127.0.0.1` and `http://[::1]` can only be used for `native` applications
- `http://127.0.0.1` and `http://[::1]` must not define any port component. URL matching must ignore the port component for those urls during auth flow.
- `https:` is required, except when the host is `127.0.0.1` or `[::1]`, in which case only `http:` is allowed.

Reply:
I don't want to duplicate these restrictions in two place. I added a reference that there are matching rules, and included them in the auth request field section (which already talks about `redirect_uri`).